target_chains/cosmwasm/tools/src/pipeline.ts

```diff
@@ -0,0 +1,181 @@
+import { readFileSync, writeFileSync, existsSync, mkdirSync } from "fs";
+import { createInterface } from "readline";
+import path from "path";
+
+// This function lets you write a question to the terminal
+// And returns the response of the user
+function readLineAsync(msg: string) {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve) => {
+    readline.question(msg, (userRes) => {
+      resolve(userRes);
+    });
+  });
+}
+
+// The stage executor is where the stage functionality is defined
+// Optionally it can take in a method `getResultOfPastStage` as a parameter
+// if it wants to access the result of the previous stages
+export type StageExecutor =
```

Comment:
probably should document that each stage should have exactly one atomic operation (like sending a transaction), because this code doesn't enforce atomicity. So if you had 2 atomic operations in one stage, then you could end up fulfilling one and failing the other.